composer.json

```diff
@@ -1,52 +1,35 @@
 {
-    "require": {
-        "php": ">=5.4.0"
-    },
-    "require-dev": {
-        "phpunit/phpunit": "*",
-        "10up/wp_mock": "*",
-        "antecedent/patchwork": "*",
-        "wordpress": "*",
-        "php-coveralls/php-coveralls": "*",
-        "squizlabs/php_codesniffer": "*",
-        "wp-coding-standards/wpcs": "*"
-    },
-    "scripts": {
-        "post-install-cmd": [
-            "\"bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs"
-        ],
-        "post-update-cmd": [
-            "\"bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs"
-        ]
-    },
-    "extra": {
-        "installer-paths": {
-            "vendor/{$name}/": [
-                "wordpress"
-            ]
-        }
-    },
-    "repositories": [
-        {
-            "type": "composer",
-            "url": "https://wpackagist.org"
-        },
-        {
-            "type": "package",
-            "package": {
-                "name": "wordpress",
-                "version": "4.9.8",
-                "type": "wordpress-plugin",
-                "dist": {
-                    "url": "https://wordpress.org/wordpress-4.9.8.zip",
-                    "type": "zip"
-                }
-            }
-        }
+  "require": {
+    "php": ">=7.2.0"
+  },
+  "require-dev": {
+    "phpunit/phpunit": "*",
+    "10up/wp_mock": "0.3.0",
+    "antecedent/patchwork": "*",
+    "php-coveralls/php-coveralls": "^2.1",
+    "squizlabs/php_codesniffer": "*",
+    "wp-coding-standards/wpcs": "*",
+    "dealerdirect/phpcodesniffer-composer-installer": "*",
+    "edsonmedina/php_testability": "dev-master",
+    "rregeer/phpunit-coverage-check": "^0.1.6",
+    "phpunit/phpcov": "^5.0"
+  },
+  "scripts": {
+    "post-install-cmd": [
```

Comment:
Add https://github.com/woocart/woocart-defaults/blob/master/grumphp.yml and "dealerdirect/phpcodesniffer-composer-installer": "*", which will set coding standard and do precheck on each commit.